tests/python_package_test/test_callback.py

```diff
@@ -20,3 +20,19 @@ def test_early_stopping_callback_is_picklable(serializer, tmp_path):
         serializer=serializer
     )
     assert callback.stopping_rounds == callback_from_disk.stopping_rounds
+
+
+@pytest.mark.parametrize('serializer', ["pickle", "joblib", "cloudpickle"])
+def test_log_evaluation_callback_is_picklable(serializer, tmp_path):
+    callback = lgb.log_evaluation(period=42)
+    tmp_file = tmp_path / "log_evaluation.pkl"
+    pickle_obj(
+        obj=callback,
+        filepath=tmp_file,
+        serializer=serializer
+    )
+    callback_from_disk = unpickle_obj(
+        filepath=tmp_file,
+        serializer=serializer
+    )
+    assert callback.period == callback_from_disk.period
```

Comment:
```suggestion
    assert callback.period == callback_from_disk.period
    assert callback.period == 42
```

Could you also add one more test against the literal value `42`?

I'm just thinking that such a check would catch the case where some refactoring accidentally resulted in argument `period` being silently ignored or not stored correctly.

Reply:
Thank you for the nice idea! Done in 617d207a0b371c7aed206baff42f18a47a2eef72.

Reply:
changes look good to me, thanks!